### PR TITLE
REF: Read excel parse refactor

### PR DESCRIPTION
**Pull request by iangainey**
This  is related to PR 58464 as a precursor to that enhancement
- [x] Precursor to closing #38937 by refactoring the parse method
- [N/A] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [N/A] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [N/A] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/io/excel/_base.py [hunk 1]
```diff
@@ -780,143 +780,195 @@ def parse(
...
-                if offset < len(data):
-                    assert isinstance(index_col, Sequence)
+        if ret_dict:
+            return output
+        else:
+            return output[last_sheetname]
 
-                    for col in index_col:
-                        last = data[offset][col]
+    def parse_sheet(
```
Comment (rhshadrach) [inline pandas/io/excel/_base.py:816]: Since this class is in the public API, we'll want to mark this internal. Can you rename to `_parse_sheet`.

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Thanks for the PR! Looks great - a rename request (I had forgotten the _ when I made the recommendation on the other PR).

File: pandas/io/excel/_base.py (hunk 1 above)
Comment (iangainey) [reply pandas/io/excel/_base.py]: Of course, I've corrected it. Thank you for taking a loot at it. Let me know if there's anything else that can be changed or improved in this PR!

**Review by rhshadrach — APPROVED** [review]
lgtm

**Comment by rhshadrach** [comment]
Thanks @iangainey!

**Pull request merged**